### PR TITLE
[WIP] Added Maxim Model

### PR DESCRIPTION
**Pull request by IMvision12**
# What does this PR do?

Ported from Sayak Paul's implementation. More information can be found here at https://github.com/sayakpaul/maxim-tf

Fixes https://github.com/keras-team/keras-cv/issues/924.
Added maxim(MAXIM: Multi-Axis MLP for Image Processing) model, Initially I was following Original [Jax](https://github.com/google-research/maxim) Implementation, but the @innat Suggested to go through @sayakpaul 's implementation [Here](https://github.com/sayakpaul/maxim-tf), till then I had almost converted flax code to TF and was similar to the sayak's implementation. Currently I have added only model architecture script.

Collaborating with @suvadityamuk
## Questions
1. As we already have TF weights on [TF-Hub](https://tfhub.dev/sayakpaul/collections/maxim/1) converted and uploaded by @sayakpaul , do we still need the conversion script?

@tanzhenyu need your review on this, Also What part of code needs to changed as per the kcv conventions!!!
@DavidLandup0

### CODE REVIEW

**Comment by tanzhenyu** [comment]
Thanks @IMvision12 for the contribution!
Re your questions-
1. Yes we should still have weights conversion script, but it could be coming from Jax or TFHub
2. Can this PR re-use the SE Layer we already have in KCV?

**Comment by DavidLandup0** [comment]
Thanks for the PR!
Since this is practically a verbatim port of @sayakpaul's implementation, he should be credited fully.

**Comment by IMvision12** [comment]
> Thanks for the PR! Since this is practically a verbatim port of @sayakpaul's implementation, he should be credited fully.

Yeah, That's totally right, the implementation belongs to @sayakpaul, so should close this PR and let him open the PR? @DavidLandup0

**Comment by IMvision12** [comment]
> Thanks @IMvision12 for the contribution! Re your questions-
> 
> 1. Yes we should still have weights conversion script, but it could be coming from Jax or TFHub
> 2. Can this PR re-use the SE Layer we already have in KCV?

Okay, should i continue with the pr?

**Comment by tanzhenyu** [comment]
> > Thanks @IMvision12 for the contribution! Re your questions-
> > 
> > 1. Yes we should still have weights conversion script, but it could be coming from Jax or TFHub
> > 2. Can this PR re-use the SE Layer we already have in KCV?
> 
> Okay, should i continue with the pr?

Yeah go ahead! (Just cite Sayak's implementation in this PR)

**Comment by IMvision12** [comment]
> > > Thanks @IMvision12 for the contribution! Re your questions-
> > > 
> > > 1. Yes we should still have weights conversion script, but it could be coming from Jax or TFHub
> > > 2. Can this PR re-use the SE Layer we already have in KCV?
> > 
> > 
> > Okay, should i continue with the pr?
> 
> Yeah go ahead! (Just cite Sayak's implementation in this PR)

How can i do it exactly?

**Comment by IMvision12** [comment]
@DavidLandup0 Any more comments?

File: keras_cv/models/maxim.py [hunk 1]
```diff
@@ -0,0 +1,1164 @@
...
+                filters=num_channels, use_bias=True, name=f"{name}_Conv_1"
+            )(x)
+            return x_down, x
+        else:
+            return x
+
+    return apply
+
+
+def UNetDecoderBlock(
```
Comment (innat) [inline keras_cv/models/maxim.py:627]: UNet decoder and encoder - would it be possible to asses if these are required for segmentation modeling part and thus it's required to have general component?

File: keras_cv/models/maxim.py [hunk 2]
```diff
@@ -0,0 +1,1164 @@
...
+            reduction=reduction,
+            name=f"{name}_channel_attention",
+        )(y)
+        x = x + y
+        return x
+
+    return apply
+
+
+def SAM(
```
Comment (innat) [inline keras_cv/models/maxim.py:114]: This might needs to be general component. 

 Multi-Stage Progressive Image Restoration.
Cited by 426 

cc. @tanzhenyu

File: keras_cv/models/maxim.py [hunk 3]
```diff
@@ -0,0 +1,1164 @@
...
+    def call(self, x, axis_one, axis_two):
+        return tnp.swapaxes(x, axis_one, axis_two)
+
+    def get_config(self):
+        config = super().get_config().copy()
+        return config
+
+
+@tf.keras.utils.register_keras_serializable("maxim")
+class Resizing(layers.Layer):
```
Comment (innat) [inline keras_cv/models/maxim.py:737]: how about `keras.layer.Resizing` ?

File: keras_cv/models/maxim.py [hunk 4]
```diff
@@ -0,0 +1,1164 @@
...
+Conv1x1 = functools.partial(layers.Conv2D, kernel_size=(1, 1), padding="same")
+ConvT_up = functools.partial(
+    layers.Conv2DTranspose, kernel_size=(2, 2), strides=(2, 2), padding="same"
+)
+Conv_down = functools.partial(
+    layers.Conv2D, kernel_size=(4, 4), strides=(2, 2), padding="same"
+)
+
+
+def CALayer(
```
Comment (innat) [inline keras_cv/models/maxim.py:39]: https://github.com/keras-team/keras-cv/blob/fca60436bf92577ade85bb734840a82965da65b7/keras_cv/layers/regularization/squeeze_excite.py#L20

File: keras_cv/models/maxim.py (hunk 3 above)
Comment (IMvision12) [reply keras_cv/models/maxim.py:737]: We can't use `keras.layer.Resizing` as don't get functionality of anti-aliasing, which comes with tf.image.resize

File: keras_cv/models/maxim.py (hunk 4 above)
Comment (IMvision12) [reply keras_cv/models/maxim.py:39]: okay will change that!

**Comment by IMvision12** [comment]
> Thanks @IMvision12 for the contribution! Re your questions-
> 
> 1. Yes we should still have weights conversion script, but it could be coming from Jax or TFHub
> 2. Can this PR re-use the SE Layer we already have in KCV?

No we can't resues SE layer becoz of this "CALayer and SqueezeAndExcite2D operation for bottleneck_filters if different, for Maxim it is filters // ratio and SqueezeAndExcite2D it is filters * ratio"

File: keras_cv/models/maxim.py (hunk 3 above)
Comment (innat) [reply keras_cv/models/maxim.py:789]: cc. @tanzhenyu
